### PR TITLE
[Java]Use gcs client instead of redis client to get session dir

### PR DESCRIPTION
**Pull request by WangTaoTheTonic**
## Why are these changes needed?

This is the final places using Redis Client directly. After this PR we can remove Redis Client and its dependencies like Jedis completely in Java lang layer.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -158,6 +158,9 @@ java/**/.classpath
 java/**/.project
 java/runtime/native_dependencies/
 
+# Cpp
+cpp/example/thirdparty/
```
Comment (jovany-wang) [inline .gitignore:162]: Great!!!

File: src/ray/gcs/gcs_client/global_state_accessor.h [hunk 2]
```diff
@@ -162,12 +162,19 @@ class GlobalStateAccessor {
...
   /// \param placement_group_name The name of placement group to look up in the GCS
-  /// Service. \return Placement group info. To support multi-language, we serialize each
+  /// Service.
+  /// \return Placement group info. To support multi-language, we serialize each
   /// PlacementGroupTableData and return the serialized string. Where used, it needs to be
   /// deserialized with protobuf function.
   std::unique_ptr<std::string> GetPlacementGroupByName(
       const std::string &placement_group_name, const std::string &ray_namespace);
 
+  /// Get value of key from GCS Service.
```
Comment (jovany-wang) [inline src/ray/gcs/gcs_client/global_state_accessor.h:172]:
```suggestion
  /// Get value of the key from GCS Service.
```

File: src/ray/gcs/gcs_client/global_state_accessor.cc [hunk 3]
```diff
@@ -262,6 +262,16 @@ std::unique_ptr<std::string> GlobalStateAccessor::GetPlacementGroupByName(
...
 }
 
+std::unique_ptr<std::string> GlobalStateAccessor::GetInternalKV(const std::string &key) {
+  std::string value;
+  Status status = gcs_client_->InternalKV().Get(key, value);
+  if (status.ok()) {
+    return std::make_unique<std::string>(value);
+  } else {
+    return nullptr;
+  }
```
Comment (jovany-wang) [inline src/ray/gcs/gcs_client/global_state_accessor.cc:272]: How about simplifying to:
return status.ok() ? std::make_unique<std::string>(value) : nullptr;

File: java/runtime/src/main/java/io/ray/runtime/RayNativeRuntime.java [hunk 4]
```diff
@@ -47,14 +46,11 @@ public RayNativeRuntime(RayConfig rayConfig) {
...
-  private void updateSessionDir() {
-    if (rayConfig.workerMode == WorkerType.DRIVER) {
-      // Fetch session dir from GCS if this is a driver.
-      RedisClient client = new RedisClient(rayConfig.getRedisAddress(), rayConfig.redisPassword);
-      final String sessionDir = client.get("session_dir", "value");
-      Preconditions.checkNotNull(sessionDir);
-      rayConfig.setSessionDir(sessionDir);
-    }
+  private void updateSessionDir(GcsClient gcsClient) {
+    // Fetch session dir from GCS if this is a driver.
```
Comment (jovany-wang) [inline java/runtime/src/main/java/io/ray/runtime/RayNativeRuntime.java:50]:
```suggestion
    // Fetch session dir from GCS.
```

**Review by jovany-wang — APPROVED** [review]
LGTM.

File: src/ray/gcs/gcs_client/global_state_accessor.cc (hunk 3 above)
Comment (WangTaoTheTonic) [reply src/ray/gcs/gcs_client/global_state_accessor.cc:272]: sure.

**Comment by jovany-wang** [comment]
A quick question: Do you plan to remove Redis client from Java in next PR right?

**Comment by WangTaoTheTonic** [comment]
> right

Yes！

**Review by wuisawesome — APPROVED** [review]

**Pull request merged**